onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorCast.cpp

```diff
@@ -28,9 +28,60 @@ class DmlOperatorCast : public DmlOperator
         castDesc.InputTensor = inputDescs.data();
         castDesc.OutputTensor = outputDescs.data();
 
-        DML_OPERATOR_DESC opDesc = { DML_OPERATOR_CAST, &castDesc };
+        if (kernelInfo.GetOutputEdgeDescription(0).tensorDataType == static_cast<MLOperatorTensorDataType>(ONNX_NAMESPACE::TensorProto_DataType_BOOL))
+        {
+            DML_OPERATOR_DESC dmlCastDesc = { DML_OPERATOR_CAST, &castDesc };
 
-        SetDmlOperatorDesc(opDesc, kernelInfo);
+            DML_ELEMENT_WISE_CLIP1_OPERATOR_DESC clipDesc = {};
+            clipDesc.InputTensor = outputDescs.data();
+            clipDesc.OutputTensor = outputDescs.data();
+            DML_SCALAR_UNION minValue;
+            minValue.Int32 = 0;
+            DML_SCALAR_UNION maxValue;
+            maxValue.Int32 = 1;
+            clipDesc.Min = minValue;
+            clipDesc.Max = maxValue;
```

Comment:
```suggestion
            clipDesc.Min.UInt8 = 0;
            clipDesc.Max.UInt8 = 1;
```
Just assign directly, and use the unsigned 8-bit one, since boolean's in ONNX are 8 bits and don't have signs.